google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/apis/ServiceAccountKeyUltimateDisplayDialog.java

```diff
@@ -0,0 +1,303 @@
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.intellij.CommonBundle;
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.RunnerAndConfigurationSettings;
+import com.intellij.execution.executors.DefaultRunExecutor;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.execution.util.EnvironmentVariable;
+import com.intellij.javaee.run.configuration.RunnerSpecificLocalConfigurationBit;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.BooleanTableCellEditor;
+import com.intellij.ui.BooleanTableCellRenderer;
+import com.intellij.ui.TableUtil;
+import com.intellij.ui.table.JBTable;
+import com.intellij.util.ui.UIUtil;
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.IntStream;
+import javax.swing.Action;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.ListSelectionModel;
+import javax.swing.border.Border;
+import javax.swing.border.EmptyBorder;
+import javax.swing.table.DefaultTableCellRenderer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Dialog visible only in Ultimate Edition that (1) confirms the download of the service account
+ * JSON key for the Google Cloud Libraries , (2) provides information on how to set the environment
+ * variables for local run and (3) allows the user to select App Engine Standard run configurations
+ * to automatically update with these environment variables.
+ */
+public class ServiceAccountKeyUltimateDisplayDialog extends DialogWrapper {
+
+  private static final Logger LOG =
+      Logger.getInstance(ServiceAccountKeyUltimateDisplayDialog.class);
+
+  private final Project project;
+  private final String gcpProjectId;
+  private final String downloadPath;
+  private JPanel mainPanel;
+  private ServiceAccountKeyDownloadedPanel commonPanel;
+  private JTable table;
+  private JLabel label;
+  private JScrollPane scrollPane;
+  private BooleanTableModel tableModel;
+
+  @VisibleForTesting public static List<RunnerAndConfigurationSettings> configurationSettingsList;
+
+  public ServiceAccountKeyUltimateDisplayDialog(
+      @Nullable Project project, @NotNull String gcpProjectId, @NotNull String downloadPath) {
+    super(project);
+    this.project = project;
+    this.gcpProjectId = gcpProjectId;
+    this.downloadPath = downloadPath;
+    init();
+    setTitle(GctBundle.message("cloud.apis.service.account.key.downloaded.title"));
+    table.setTableHeader(null);
+
+    if (tableModel.getRowCount() == 0) {
```

Comment:
This looks slightly mysterious, would it be more clear if we just call `getAppEngineStandardConfigurationSettingsList` directly to check we don't have any configs available?